### PR TITLE
deposit form: layout improvements

### PR DESCRIPTION
**Pull request by jennur**
- Adjusted according to updates in PR https://github.com/inveniosoftware/react-invenio-deposit/pull/472

### CODE REVIEW

File: invenio_app_rdm/theme/assets/semantic-ui/less/invenio_app_rdm/theme/collections/form.overrides [hunk 1]
```diff
@@ -60,14 +56,32 @@
...
           position: absolute;
           top: 1rem;
           right: 1.3rem;
+          margin-top: 0;
+        }
+      }
+    }
+
+    .additional-descriptions {
+      margin-top: -1rem;
```
Comment (kpsherva) [inline invenio_app_rdm/theme/assets/semantic-ui/less/invenio_app_rdm/theme/collections/form.overrides:69]: is it possible to do without negative margins ? they create a lot of issues, especially when used with SUI

File: invenio_app_rdm/theme/assets/semantic-ui/less/invenio_app_rdm/theme/collections/form.overrides [hunk 2]
```diff
@@ -60,14 +56,32 @@
...
+
+      .ui.icon.button.close-btn {
+        margin-top: 0;
+      }
+
+      .ui.grid {
+        margin: 1rem 0 0 0;
```
Comment (kpsherva) [inline invenio_app_rdm/theme/assets/semantic-ui/less/invenio_app_rdm/theme/collections/form.overrides:76]: seems to be related to the negative margin ? why is it needed ?

File: invenio_app_rdm/theme/assets/semantic-ui/less/invenio_app_rdm/theme/collections/form.overrides [hunk 3]
```diff
@@ -60,14 +56,32 @@
...
+
+        .row {
+          padding-bottom: 2rem;
```
Comment (kpsherva) [inline invenio_app_rdm/theme/assets/semantic-ui/less/invenio_app_rdm/theme/collections/form.overrides:79]: there is a utility class you could use instead `rel-pb-2` unless it is meant to be applied to every row inside of the additional-descriptions

**Comment by jennur** [comment]
Please also review https://github.com/inveniosoftware/react-invenio-forms/pull/128 which is related to this PR.

File: invenio_app_rdm/theme/assets/semantic-ui/less/invenio_app_rdm/theme/collections/form.overrides (hunk 3 above)
Comment (jennur) [reply invenio_app_rdm/theme/assets/semantic-ui/less/invenio_app_rdm/theme/collections/form.overrides:82]: Agreed to keep. Discussed in person.

**Pull request merged**